src/posix/fs.cpp

```diff
@@ -1950,7 +1950,18 @@ bool PosixDirAccess::dnext(string* path, string* name, bool followsymlinks, node
         {
             path->append(d->d_name);
 
-            if (followsymlinks ? !stat(path->c_str(), &statbuf) : !lstat(path->c_str(), &statbuf))
+            bool statOk = !lstat(path->c_str(), &statbuf);
+            if (statOk && S_ISLNK(statbuf.st_mode) && followsymlinks)
```

Comment:
very minor perf improvement:

```suggestion
            if (followsymlinks && statOk && S_ISLNK(statbuf.st_mode))
```

Reply:
will make this change shortly, thanks